src/lib/metrics-gauge.ts

```diff
@@ -37,17 +37,39 @@ export class DbMetricsMonitor {
         return Array.isArray(value) ? value : [value];
     }
 
+    private async fetch<T, L extends string>(
+        definition: GaugeDefinition<T, L>,
+    ): Promise<MetricValue<L>[]> {
+        const result = await definition.query();
+        if (
+            result !== undefined &&
+            result !== null &&
+            (!Array.isArray(result) || result.length > 0)
+        ) {
+            const resultArray = this.asArray(definition.map(result));
+            resultArray
+                .filter((r) => !Number.isInteger(r.value))
+                .forEach((r) => {
+                    this.log.warn(
+                        `Invalid value for ${definition.name}: ${r.value}. Value must be an integer.`,
+                    );
+                });
+            return resultArray.filter((r) => Number.isInteger(r.value));
+        }
+        return [];
+    }
```

Comment:
Encapsulate transformation from the query result (which can either be a single result or an array), into an array of valid metrics for prometheus